src/server.c

```diff
@@ -2441,6 +2441,12 @@ void initServer(void) {
     server.reply_buffer_resizing_enabled = 1;
     resetReplicationBuffer();
 
+    /* Make sure the locale is set on startup based on the config file. */
+    if (setlocale(LC_COLLATE,server.locale_collate) == NULL) {
+        serverLog(LL_WARNING, "Failed to configure LOCALE for invalid locale name.");
+        exit(1);
+    }
```

Comment:
i found a odd issue:

if i set `ulimit -Sv 100000`, then it is unable to start the server (missing some logs).
with the default locale-collate value ""
```
[root@binblog redis]# src/redis-server redis.conf --locale-collate ""
16286:C 21 Jun 2023 11:37:07.391 * oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
16286:C 21 Jun 2023 11:37:07.391 * Redis version=255.255.255, bits=64, commit=07c14672, modified=1, pid=16286, just started
16286:C 21 Jun 2023 11:37:07.392 * Configuration loaded
16286:M 21 Jun 2023 11:37:07.392 # Failed to configure LOCALE for invalid locale name.
```

and it is ok the set a another value:
```
[root@binblog redis]# src/redis-server redis.conf --locale-collate "C"

# is ok, in this case, using config set
127.0.0.1:6379> config get locale-collate
1) "locale-collate"
2) "C"
127.0.0.1:6379> config set locale-collate ""
(error) ERR CONFIG SET failed (possibly related to argument 'locale-collate') - Invalid locale name
```

Is there a better way to print errors here (i support it is somehow an OOM error),
or rather not let the server exit